lib/dgram.js

```diff
@@ -78,7 +78,7 @@ exports._createSocketHandle = function(address, port, addressType, fd, flags) {
 function Socket(type, listener) {
   EventEmitter.call(this);
 
-  if (typeof type === 'object') {
+  if (type !== null && typeof type === 'object') {
```

Comment:
@claudiorodriguez I’ll try to look it up/benchmark it, all I could remember right now is https://github.com/emberjs/ember.js/pull/14118 … might be a different case with `!== null`, but even so I think I’d find the `typeof`-first variants more readable.